huff0/_generate/gen.go

```diff
@@ -0,0 +1,366 @@
+package main
+
+//go:generate go run gen.go -out decompress_amd64.s -stubs delme.go -pkg=huff0
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	_ "github.com/klauspost/compress"
+
+	. "github.com/mmcloughlin/avo/build"
+	"github.com/mmcloughlin/avo/buildtags"
+	"github.com/mmcloughlin/avo/gotypes"
+	. "github.com/mmcloughlin/avo/operand"
+	"github.com/mmcloughlin/avo/reg"
+)
+
+func main() {
+	flag.Parse()
+	out := flag.Lookup("out")
+	os.Remove(filepath.Join("..", out.Value.String()))
+	stub := flag.Lookup("stubs")
+	if stub.Value.String() != "" {
+		os.Remove(stub.Value.String())
+		defer os.Remove(stub.Value.String())
+	}
+
+	Constraint(buildtags.Not("appengine").ToConstraint())
+	Constraint(buildtags.Not("noasm").ToConstraint())
+	Constraint(buildtags.Term("gc").ToConstraint())
+	Constraint(buildtags.Not("noasm").ToConstraint())
```

Comment:
You can use `ConstraintExpr` and just provide a string (in the old `+build` syntax).